SingularityService/src/main/java/com/hubspot/singularity/data/AbstractMachineManager.java

```diff
@@ -52,9 +55,16 @@ private String getHistoryPath(String objectId) {
   }
 
   public List<T> getObjects() {
-    return getObjects(getRoot());
+    List<T> fromCache = getObjectsFromLeaderCache();
+    if (fromCache != null) {
```

Comment:
There look to be at least two places where this'll never be null: `SingularityLeaderCache#getRacks()` and `SingularityLeaderCache#getSlaves()` do we want to check this for emptiness as well as nullity, and fall back to `getObjectsNoCache()` in either case?

Reply:
ah, good point, maybe I need the `isActive` check in the leader cache to return null instead